restli-resources/src/main/java/com/linkedin/metadata/restli/BaseAspectRoutingResource.java

```
@@ -0,0 +1,163 @@
+   * @return True if aspectClasses contains routing aspect class.
+   */
+  private boolean containsRoutingAspect(Set<Class<? extends RecordTemplate>> aspectClasses) {
+    return aspectClasses.stream().anyMatch(aspectClass -> aspectClass.equals(_routingAspectClass));
+  }
+
+  /**
+   * Get entity value from local DAO decorated with specified aspect classes.
+   * @param id identifier of the entity.
+   * @param aspectClasses Aspects to be decorated on the entity
+   * @return Entity decorated with specified aspect classes.
+   */
+  private VALUE getValueFromLocalDao(KEY id, Set<Class<? extends RecordTemplate>> aspectClasses) {
+    final URN urn = toUrn(id);
+    final VALUE value = getInternal(Collections.singleton(urn), aspectClasses).get(urn);
+    if (value == null) {
+      throw RestliUtils.resourceNotFoundException();
+    }
+    return value;
+  }
+
+  /**
+   * Merge routing aspect value from GMS into entity value retrieved from Local DAO.
+   * @param valueFromLocalDao Entity value retrieved from Local DAO
+   * @param aspectFromGms Entity value retrieved from GMS
+   * @return Merge entity value which will contain routing aspect value
+   */
+  private VALUE merge(@Nullable VALUE valueFromLocalDao, @Nullable ROUTING_ASPECT aspectFromGms) {
+    final String setterMethodName = "set" + getRoutingAspectFieldName();
+
+    if (valueFromLocalDao == null) {
+      try {
+        valueFromLocalDao = _valueClass.getDeclaredConstructor().newInstance();
+      } catch (Exception e) {
+        throw new RestLiServiceException(HttpStatus.S_500_INTERNAL_SERVER_ERROR,
+            "Failed to create new instance of class " + _valueClass.getCanonicalName());
+      }
+    }
+
+    Method setter = null;
+    try {
+      setter = valueFromLocalDao.getClass().getMethod(setterMethodName, _routingAspectClass, SetMode.class);
+    } catch (NoSuchMethodException e) {
+      throw new RestLiServiceException(HttpStatus.S_500_INTERNAL_SERVER_ERROR,
```

Do you want to swallow the exception message here?